public/app/plugins/panel/table/module.tsx

```diff
@@ -77,6 +77,26 @@ export const plugin = new PanelPlugin<PanelOptions, TableFieldOptions>(TablePane
           },
           defaultValue: defaultPanelFieldConfig.displayMode,
         })
+        .addRadio({
+          path: 'gaugeLabels',
+          name: 'Display gauge labels',
+          description: 'How labels are displayed on guages',
+          settings: {
+            options: [
+              { label: 'Always', value: 'always' },
+              { label: 'Never', value: 'never' },
+              { label: 'Overlay', value: 'overlay' },
+            ],
+          },
+          showIf: (cfg) => {
+            return (
+              cfg.displayMode === TableCellDisplayMode.BasicGauge ||
+              cfg.displayMode === TableCellDisplayMode.GradientGauge ||
+              cfg.displayMode === TableCellDisplayMode.LcdGauge
+            );
+          },
+          defaultValue: defaultPanelOptions.gaugeLabels,
+        })
```

Comment:
think we should do this https://github.com/grafana/grafana/issues/31315 before so each cell mode becomes a object that can more sub options.

Reply:
That makes sense and definitely agree. Going to keep this in draft and will circle back for that 😄